CalendarFXView/src/main/java/com/calendarfx/view/DateControl.java

```diff
@@ -209,21 +209,24 @@ protected DateControl() {
         setEntryDetailsPopOverContentCallback(param -> new EntryPopOverContentPane(param.getPopOver(), param.getDateControl(), param.getEntry()));
 
         /*
-         * The default calendar provider returns the first calendar from the
-         * first source.
+         * The default calendar provider returns the first calendar which is visible and not read-only.
          */
         setDefaultCalendarProvider(control -> {
             List<CalendarSource> sources = getCalendarSources();
-            if (sources != null && !sources.isEmpty()) {
-                CalendarSource s = sources.get(0);
-                List<? extends Calendar> calendars = s.getCalendars();
-                if (calendars != null && !calendars.isEmpty()) {
-                    for (Calendar c : calendars) {
-                        if (!c.isReadOnly() && isCalendarVisible(c)) {
-                            return c;
+            if (sources != null) {
+                boolean calendarsDefined = false;
+                for (CalendarSource s : sources) {
+                    List<? extends com.calendarfx.model.Calendar> calendars = s.getCalendars();
```

Comment:
Can you please change so that the code does not use the fully qualified classname?